desc/plotting.py

```diff
@@ -626,8 +630,28 @@ def plot_2d(
     plot_axes = _get_plot_axes(grid)
     if len(plot_axes) != 2:
         return ValueError(colored("Grid must be 2D", "red"))
-
-    data, label = _compute(eq, name, grid, kwargs.pop("component", None))
+    component = kwargs.pop("component", None)
+    if name != "Bn":
+        data, label = _compute(
+            eq,
+            name,
+            grid,
+            component=component,
+        )
+    else:
+        field = kwargs.pop("field", None)
+        errorif(
+            field is None,
+            ValueError,
+            "If Bn is entered as the variable to plot, a magnetic field"
+            " must be provided.",
+        )
+        field_grid = kwargs.pop("field_grid", None)
+        with warnings.catch_warnings():
+            warnings.simplefilter("ignore")
+            data, _ = field.compute_Bnormal(eq, eval_grid=grid, source_grid=field_grid)
+        data = data.reshape((grid.num_theta, grid.num_rho, grid.num_zeta), order="F")
+        label = r"$|B_n| ~(\mathrm{T})$"
```

Comment:
similar to the above comment, change this label to `r"$|\mathbf{B}\cdot\hat{n}| ~(\mathrm{T})$"` to better match our existing compute quantities